programs/psinode/tests/test_psibase.py

```diff
@@ -96,8 +96,8 @@ def test_upgrade(self, cluster):
         a = cluster.complete(*testutil.generate_names(1))[0]
         a.boot(packages=['Minimal', 'Explorer', 'Sites'])
 
-        foo10 = TestPackage('foo', '1.0.0').depends('Sites').service('foo', data={'file1.txt': 'original', 'file2.txt': 'deleted'})
-        foo11 = TestPackage('foo', '1.1.0').depends('Sites').service('foo', data={'file1.txt': 'updated', 'file3.txt': 'added'})
+        foo10 = TestPackage('foo', '1.0.0').depends('Sites').service('foo', data={'file1.png': 'original', 'file2.png': 'deleted'})
+        foo11 = TestPackage('foo', '1.1.0').depends('Sites').service('foo', data={'file1.png': 'updated', 'file3.png': 'added'})
```

Comment:
It would probably be better to ensure that it's decoded. I don't think these are valid PNGs:

"The gzip and deflate transfer-encodings are automatically decoded for you.

The br transfer-encoding is automatically decoded for you if a Brotli library like [brotli](https://pypi.org/project/brotli) or [brotlicffi](https://pypi.org/project/brotlicffi) is installed." (from the [requests docs](https://docs.python-requests.org/en/latest/user/quickstart/#binary-response-content))

Reply:
We're actually cheating a bit here. The server is supposed to check `Accept-Encoding` before sending a compressed response.

Reply:
Haha yeah this was definitely just a hack to trick the tooling into not compressing the file. 

Re: `Accept-Encoding`, looks like what we are doing is technically not cheating:
```
A request without an Accept-Encoding header field implies that the
   user agent has no preferences regarding content-codings.  Although
   this allows the server to use any content-coding in a response, it
   does not imply that the user agent will be able to correctly process
   all encodings.
```
[RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.4)

Although I think the implementation is incomplete, `Sites` should verify that the requests either has no `Accept-Encoding`, or that it explicitly specifies that `br` is accepted. In case the request has `Accept-Encoding` with a value that is not `br`, I think the server should return `406 Not Acceptable`. Agree?

Reply:
Python *does* send `Accept-Encoding': gzip, deflate`.

Reply:
> > If a non-empty Accept-Encoding header field is present in a request and none of the available representations for the response have a content coding that is listed as acceptable, the origin server SHOULD send a response without any content coding unless the identity coding is indicated as unacceptable.

For Python I will install the brotli lib so it adds it to the Accept-Encoding. But in general I guess we do have to cheat on this part. For cases where the client explicitly doesn't support Brotli all we can do is 406 it... Unless we want compression to happen server side, which I don't think we do.

Reply:
> Unless we want compression to happen server side at the time of the request, which... I don't think we do?

Fallback *de*compression on the server side would be more viable.

Reply:
Fallback decompression implemented for Brotli